hooks/expand_show_tree.py

```diff
@@ -0,0 +1,185 @@
+The default implementation expands this path on the basis of the context of ``bundle_obj``
+And will construct an inheritance model as per the order shown below.
+
+Entries in the lookup path are arranged so the inheritance is as follows:
+
+    - show
+    - show-workarea
+    - sequence
+    - sequence-workarea
+    - shot
+    - shot-workarea
+
+At each level, hook paths are searched for in the following order:
+
+    - <filename>
+    - <filename>_<role>
+    - <filename>_<role>_<subrole>
+
+This would typically not need to be customized.
+
+In case customization is required, the hook is passed the app/engine/framework
+that issued the original request - this gives access to configuration,
+app methods, environment etc. and should allow for some sophisticated
+introspection inside the hook.
+"""
+
+import os
+
+import sgtk
+from sgtk import dd_xplatform_utils
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class ExpandShowTree(HookBaseClass):
+    
+    def execute(self, path, bundle_obj, user_subdir="etc", show_subdir="SHARED/etc",
+                extra_subdir="sgtk/hooks", **kwargs):
+        """
+        Handle expansion of the given path on show tree, depending on context issued from an app, framework or engine.
+        
+        :param path: path to create
+        :param bundle_obj: object requesting the creation. This is a legacy
+                              parameter and we recommend using self.parent instead.
+        :param user_subdir: Subdirectories to use for resolving work-area hooks, can be colon separated.
+        :param show_subdir: Subdirectories to use for resolving show-tree hooks, can be colon separated.
+        :param extra_subdir: additional subdirectories to be included in the hierarchy, CAN'T be colon separated.
+
+        :return: A colon separated list of paths in the order of Test_User:Shot:Seq:Project,
+        code that calls it can decide what to do with that paths.
+        """
+
+        user_subdir_list = user_subdir.split(os.pathsep)
+        show_subdir_list = show_subdir.split(os.pathsep)
+
+        if extra_subdir:
+            user_subdir_list = [os.path.join(x, extra_subdir) for x in user_subdir_list]
+            show_subdir_list = [os.path.join(x, extra_subdir) for x in show_subdir_list]
+
+        # this is (:) separated list of paths
+        show_paths = self.expand_tree(bundle_obj, user_subdir_list, show_subdir_list, path)
+
+        # return colon separated list of paths, to keep the inheritance model intact.
+        return show_paths
+
+        # we need to return the first path since that is the override we have configured
+        # Seq:Shot:Project
+        # if paths_list:
+        #     return paths_list[0]
+
+        # none of the paths exist! the hook was configured without actually populating the show with the hook.
+        # return ''
+
+    @staticmethod
+    def expand_tree(bundle_obj, user_subdir_list, show_subdir_list, path):
```

Comment:
Optional, but since you're already supporting most of the Preferences path expansion logic, what about adding location subdir support as well?

Reply:
Preferences module has <site> as a part of the file-name resolution, we will be diverging from that since for us it will make sense to add the <site> at the base of the search directories, something like.
`SHARED/etc/<site>/sgtk/hooks` or `user/work.$USER/etc/<site>/sgtk/hooks`. 

Are you anticipating adding this to search paths might benefit us?

Reply:
I think it makes sense to have site always succeed etc...in other words, option 1

Reply:
Actually, I take it back. We should keep it consistent with Preferences